### PR TITLE
Allow @types/react 18 as a peer dep for @backstage/theme

### PR DESCRIPTION
**Pull request by fshowalter**
## Hey, I just made a Pull Request!

Looks like this was missed during the React 18 upgrade. 

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by tudi2d — CHANGES_REQUESTED** [review]
Thanks! Interestesting - not sure if this was on purpose - Patrik will know.

You would need to attach a `patch` changelog for the package & also commit the `yarn.lock` changes - thanks!:)

**Comment by fshowalter** [comment]
> Thanks! Interestesting - not sure if this was on purpose - Patrik will know.
> 
> You would need to attach a `patch` changelog for the package & also commit the `yarn.lock` changes - thanks!:)

Done.

**Review by Rugvip — CHANGES_REQUESTED** [review]
Looks good, thank you! 👍 

This does apply to a lot more packages though, although as regular dependencies. Do you want to update this PR to include all packages that currently declare a 16^ | 17^ range, since it's best to do them all at once?

**Comment by fshowalter** [comment]
@Rugvip I can do that. 😄

**Comment by fshowalter** [comment]
@Rugvip done.

File: .changeset/olive-boats-agree.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+'@backstage/plugin-techdocs-react': patch
+'@backstage/plugin-techdocs': patch
+'@backstage/plugin-user-settings': patch
+'@backstage/plugin-vault': patch
+'@backstage/plugin-xcmetrics': patch
+'@backstage/test-utils': patch
+'@backstage/theme': patch
+---
+
+Widen @types/react range to include 18.
```
Comment (Rugvip) [inline .changeset/olive-boats-agree.md:93]:
```suggestion
Widen `@types/react` dependency range to include version 18.
```

**Review by Rugvip — APPROVED** [review]
Sweet, thank you! 👍 

E2E test failures are unrelated, gonna :shipit:

**Pull request merged**